matplotlib2tikz/legend.py

```diff
@@ -181,6 +187,17 @@ def draw_legend(data, obj):
             alignment = None
             break
 
+    # Set color of lines in legend
+    data['legend colors'] = []
+    for handle in obj.legendHandles:
+        try:
+            data, legend_color, _ = mycol.mpl_color2xcolor(data,
+                                                           handle.get_color())
+            data['legend colors'].append('\\addlegendimage{no markers, %s}\n'
+                                         % legend_color)
+        except:
```

Comment:
Why do we need an exception type? Now we can catch any exception :D 
I introduced this try statement, because mycol.mpl_color2xcolor(data, handle.get_color()) fails in case of polar plots. I don't know wether it is necessary to account such an case. Perhaps, we can skip this again in order to keep it simple.